core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -663,7 +673,7 @@ class SnappyParser(session: SnappySession)
     SELECT ~ (DISTINCT ~> trueFn).? ~
     (namedExpression + commaSep) ~
     (FROM ~ ws ~ relations).? ~
-    (WHERE ~ expression).? ~
+    (WHERE ~ (test(setTokenize) ~ expression ~ test(setNoTokenize))).? ~
```

Comment:
Instead of "test" for setting the values, a cleaner way will be to change those methods to be Rule0 that will do the setting in its action like:

```scala
protected final def enableTokenization: Rule0 = rule {
  MATCH ~> {() => ...}
}
```

Then you can just use those as regular rules without the artificial test().